truffle/test/controller_deployment_interaction.js

```diff
@@ -0,0 +1,330 @@
+/* -------------------------------------------------------------------------- */
+
+const TuiChainController = artifacts.require("TuiChainController");
+const TuiChainMarket = artifacts.require("TuiChainMarket");
+const TuiChainLoan = artifacts.require("TuiChainLoan");
+
+const DaiMock = artifacts.require("DaiMock");
+
+/* -------------------------------------------------------------------------- */
+
+// Sequence of tests for the deployment and interaction with Controller Contract
+// This tests represent possible interactions from the backend with controller contract
+contract("Controller Deployment and Interaction", function (accounts) {
+
+
+  // market fee, is equivalent to 10% of a DAI
+  const marketFeeAttoDaiPerNanoDai = BigInt(Math.pow(10,8));
+
+  /* -------------------------------------------------------------------------- */
+
+  // variable which represents the deployed DAI mock
+  let daiMock = null;
+
+  // variable which represents the deployed controller contract
+  let tuiChainController = null;
+
+  // variable which represents the deployed market contract
+  let tuiChainMarket = null;
+
+  // variable which represents the a deployed loan contract
+  let tuiChainLoan = null;
+
+  // variable which represents the loan object
+  let loanObject = null;
+
+  /* -------------------------------------------------------------------------- */
+
+  // runs once before the first test
+  before( async () => {
+
+    // deploy contract which mocks DAI contract
+    daiMock = await DaiMock.new();
+
+    // deploy controller contract
+    tuiChainController = await TuiChainController.new(daiMock.address, accounts[0], marketFeeAttoDaiPerNanoDai);
+
+    // get contract from an address with at() function
+    tuiChainMarket = await TuiChainMarket.at(await tuiChainController.getMarket());
+
+    // start every account with 1000 DAI
+    accounts.forEach( account => {
+      daiMock.mint(account, BigInt(1000 * Math.pow(10,18)));
+    });
+  
+  });
+
+  // runs before every test
+  beforeEach( async () => {
+
+    loanObject = {
+      _feeRecipient: accounts[0],
+      _loanRecipient: accounts[1],
+      _secondsToExpiration: 60, // 1 minute
+      _fundingFeeAttoDaiPerDai: BigInt(Math.pow(10,17)), // 10% fee
+      _paymentFeeAttoDaiPerDai: BigInt(Math.pow(10,17)), // 10% fee
+      _requestedValueAttoDai: BigInt(1000 * Math.pow(10,18)) // 1000 DAI
+    };
+
+  });
+
+  /* -------------------------------------------------------------------------- */
+
+  it("Fail to change market fee if not the owner", async function () {
+
+    const newFee = BigInt(2*Math.pow(10,8));
+    
+    try {
+
+      await tuiChainController.setMarketFee(newFee, {from: accounts[1]});
+
+    } catch(e) {
+
+      return assert(e.message.includes('caller is not the owner'));
```

Comment:
Is the return doing anything? Ditto for all other occurrences.

Reply:
Not in this case, because I forgot other return outside the catch block.
But it needs this return because we can stop the test right there when the wanted result is reached.

Reply:
I will add the missing return.